torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2235,6 +2244,20 @@ def create_symbolic_sizes_strides_storage_offset(
         We try our best to express stride in terms of the sizes, so as to not
         introduce new symbolic variables.
         """
+
+
+        # Dynamo may want to wrap FakeTensors with SymInt sizes up e.g. make_fx(opt_f(), tracing_mode="symbolic").
+        # We create symbols in shape_env using the backed hints behind SymInt.
+        # Case 1: when SymInt is backed, dynamo can proceed with FakeTensors that have concrete shape.
+        # produce_guards will trigger specializations on the outer stuff
+        # Case 2: when the SymInt is unbacked, we will throw an data dependent error in require_hint().
+        def maybe_specialize_sym_int_with_hint(maybe_sym) -> int:
+            assert isinstance(maybe_sym, (int, torch.SymInt))
+            return maybe_sym if isinstance(maybe_sym, int) else maybe_sym.node.require_hint()
+
+        ex_size = pytree.tree_map(maybe_specialize_sym_int_with_hint, tuple(ex.size()))
```

Comment:
There isn't any reason to do a tree_map here, just do a list comprehension for speed.